core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1815,15 +1817,59 @@ public void onSetPlaybackSpeed(float speed) {
             setSpeed(speed);
         }
 
+        private boolean playStateBeforeClick = false;
+        private long lastClickTime = 0;
+        private Handler clickHandler = new Handler(Looper.getMainLooper());
+        private int clickCount = 0;
+
+        private Runnable clickRunnable = new Runnable() {
+            @Override
+            public void run() {
+                if (clickCount == 1) {
+                    if (playStateBeforeClick) {
+                        onPause();
+                    } else {
+                        onPlay();
+                    }
```

Comment:
Why not just call `handleKeycode()` here? That should make it more consistent with actual hardware buttons

Reply:
Changed as suggested. Thanks